src/services/tx/txSender.ts

```diff
@@ -57,7 +59,22 @@ export const createTx = async (txParams: SafeTransactionDataPartial, nonce?: num
   // Set the recommended nonce and safeTxGas if not provided
   if (nonce === undefined) {
     const chainId = await safeSDK.getChainId()
-    const estimation = await estimateSafeTxGas(String(chainId), safeSDK.getAddress(), txParams)
+    const safeAddress = safeSDK.getAddress()
+    let estimation: SafeTransactionEstimation
+    try {
+      estimation = await estimateSafeTxGas(String(chainId), safeAddress, txParams)
+    } catch (e) {
+      try {
+        /* If the initial transaction data causes the estimation to fail
+         we retry the request with a cancellation transaction to get the
+         recommendedNonce even if the original transaction will likely fail */
+        const cancellationTxParams = { ...txParams, data: EMPTY_DATA, to: safeAddress, value: '0' }
+        estimation = await estimateSafeTxGas(String(chainId), safeAddress, cancellationTxParams)
+      } catch (e) {
+        logError(Errors._616, (e as Error).message)
+        return safeSDK.createTransaction({ safeTransactionData: txParams })
```

Comment:
That's code duplication...

Reply:
Totally. Next commit fixes it. Thanks for the thorough review